ValidationLibrary.AzureFunctions/RepositoryValidator.cs

```diff
@@ -27,38 +27,39 @@ public RepositoryValidator(ILogger<RepositoryValidator> logger, IGitHubClient gi
         }
 
         [FunctionName("RepositoryValidator")]
-        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)]HttpRequestMessage req)
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req)
         {
             try
             {
                 _logger.LogDebug("Repository validation hook launched");
-                if (req.Content == null)
+                if (req == null || req.Content == null)
                 {
-                    throw new ArgumentNullException("Request content was null. Unable to retrieve parameters.");
+                    throw new ArgumentNullException(nameof(req), "Request content was null. Unable to retrieve parameters.");
                 }
 
-                var content = await req.Content.ReadAsAsync<PushData>();
+                var content = await req.Content.ReadAsAsync<PushData>().ConfigureAwait(false);
                 ValidateInput(content);
 
                 _logger.LogInformation("Doing validation. Repository {owner}/{repositoryName}", content.Repository?.Owner?.Login, content.Repository?.Name);
 
-                await _validationClient.Init();
-                var report = await _validationClient.ValidateRepository(content.Repository.Owner.Login, content.Repository.Name, false);
+                await _validationClient.Init().ConfigureAwait(false);
+                var report = await _validationClient.ValidateRepository(content.Repository.Owner.Login, content.Repository.Name, false).ConfigureAwait(false);
 
                 _logger.LogDebug("Sending report.");
-                await ReportToGitHub(report);
-                await PerformAutofixes(report);
+                await ReportToGitHub(report).ConfigureAwait(false);
+                await PerformAutofixes(report).ConfigureAwait(false);
                 _logger.LogInformation("Validation finished");
                 return new OkResult();
             }
             catch (Exception exception)
             {
+                req?.Dispose();
```

Comment:
This could go to finally block an catch (Exception ..) -> catch (Exception ex) when (ex is ... || ex is ...)

Reply:
Ah, this was a leftover from my reseach on that `OutOfMemoryException` stuff, I don't think this is even needed.